Add `label` (#111)
hadley/data-dict.yaml: site/spec.md

* `$version` (required): the version of the `data-dict.yaml` spec the document conforms to. Currently `0.1.0`. While the spec is pre-1.0, breaking changes are expected, but once the spec stabilises at 1.0, breaking changes will always increment at least the minor version.
* `$learn_more` (optional, but recommended): a URL where readers can learn about the `data-dict.yaml` format, so that people and tools meeting the file for the first time can find out what it is. Use <http://data-dict.tidyverse.org/>. Omitting it is valid, but a validator will emit a warning rather than an error (see [Validation](validation.md)).

> The descriptive keys — `name`, `label`, `description`, and `details` — identify and document the dataset as a whole. All four are optional here, and work the same way at every level of the dictionary; see [Name, label, description & details](#name-label-description--details) for their full meaning. For the dataset, `name` is a terse identifier (e.g. `foodbank`) and `label` its human-readable title.

Reviewer: I this isn't part of this PR, but I have a hard time know when I might use `description` vs `details`. I might make an issue about this though.

Author: Yeah, me too 😄

#### Name, label, description & details

`name`, `label`, `description`, and `details` document a dataset, table, or column, from terse to expansive. They mean the same thing at every level:

* `name` identifies the thing. For a table or column it's an identifier matched against the underlying data, so it must be non-empty and unique (a table within the dictionary, a column within its table). For the dataset it's just a short, machine-friendly id (e.g. `foodbank`) with no constraints. It's the only one of the four that is ever required.
* `label` is a short, human-readable title, useful when the `name` is terse or technical (e.g. `FoodData Central ID` for `fdc_id`). Plain text (no markdown), typically a few words, it stands in for the `name` in user interfaces.
* `description` is a short summary, typically a few sentences or at most a paragraph. May contain markdown, and is displayed in user interfaces. A good table description answers two questions — **what's the grain?** (what does a row represent, e.g. "each row is a food item") and **what's the population?** (what's been included or filtered out, e.g. "only completed orders from 2020 onwards").

> * `details` is a free-text note of any length: the place to carefully record everything else, such as "here be dragons" assumptions, known weak spots, surprising calculations, and how the data was collected or constructed.

Reviewer: I don't know what "here be dragons" means in this context. Maybe?

```suggestion
* `details` is a free-text note of any length: the place to carefully record everything else, such as assumptions about potential unknowns, known weak spots, surprising calculations, and how the data was collected or constructed.
```

> Every field but `name` is optional at every level.

Reviewer: Related to the "strictness" checks I mentioned in other PRs/issues, I think label should never be optional but understand that might put a big burden to pass the checks for many existing datasets. Could `label` be included as a "next level required" field? A bit like how Cargo clippy has different levels of linting?

Author: I think the problem with making it required is that for many datasets you'll end with pretty redundant labels. But I made a note in #39 that this might be an optional check that folks want to turn on.
